osh/hub/scripts/osh-retention

```diff
@@ -26,96 +25,86 @@ from osh.hub.scan.models import TaskResultsRemoval  # noqa: E402
 
 logger = logging.getLogger("osh.hub.scripts.osh-retention")
 
-tasks_to_delete = []
-today = date.today()
 
+# Functions to determine membership for given retention setting
 
-def delete_tasks_results(task_list):
-    for task, retention_policy_setting in task_list:
-        try:
-            path_to_delete = Task.get_task_dir(task.id)
-            # if directory exists, we delete it. if not, we mark it as deleted as
-            # the directory already doesn't exist
-            if os.path.isdir(path_to_delete):
-                shutil.rmtree(path_to_delete)
-                logger.debug('The following directory was deleted: %s', path_to_delete)
-        except OSError as e:
-            logger.error("Error deleting task's directory: %s", e)
-            continue
+def is_failed(task):
+    return task.is_failed()
 
-        # if the tasks results are deleted, we update the database field,
-        # so we don't try to delete them again.
-        TaskResultsRemoval.objects.create(task=task, reason=retention_policy_setting)
 
+def is_personal(task):
+    # If the username does not contain /, @ or -, assume that the task is personal.
+    return not re.search('/@-', task.owner.username)
 
-# in order to identify personal tasks, we are using the owner field. If the owner doesn't
-# contain any /, @ and - we assume that it is a personal task.
-def identify_personal_task(owner):
-    # we could use regexps here if we want to make this more complex but for this specific case
-    # built-in string methods are enough
-    return "/" not in owner and "@" not in owner and "-" not in owner
 
+ELIGIBILITY_FUNCTIONS = {
+    # "setting name": function(task),
+    "FailedStatus": is_failed,
+    "PersonalScan": is_personal,
+}
 
-# if the retention time of the task has passed, we add the task to the be-deleted list
-def add_task_to_deletion_list(task, retention_policy_setting, calculated_time):
-    if retention_policy_setting is not None and calculated_time is not None and calculated_time <= today:
-        tasks_to_delete.append([task, retention_policy_setting])
-        logger.debug('The following task results will be deleted: %s', task.id)
+
+def delete_task_results(task, reason):
+    try:
+        task_dir = Task.get_task_dir(task.id)
+        shutil.rmtree(task_dir)
+        logger.debug('The following directory was deleted: %s', task_dir)
+    except FileNotFoundError:
+        # proceed silently when the task directory is missing
+        pass
+
+    except OSError as e:
+        logger.error("Error deleting task directory: %s", e)
+        return
+
+    # mark the task as processed
+    TaskResultsRemoval.objects.create(task=task, reason=reason)
 
 
-# we obtain the different retention policy settings from the database
-def get_retention_policy_parameter(parameter):
+def get_setting(name):
     try:
-        obtained_parameter = RetentionPolicySetting.objects.get(category=parameter)
+        return RetentionPolicySetting.objects.get(name=name)
```

Comment:
if we want to query based on `name`, we better make the field unique and non-nullable. otherwise there might be `MultipleObjectReturned` or other weird behavior here.